enhancements/monitoring/monitoring-stack-operator.md

```diff
@@ -0,0 +1,490 @@
+    * We need to implement Rules and Alerting API as well as Alert routing on RHOBS
+    * A-SRE, Observatorium team’s available bandwidth can be non-starter to implement tenant features on the Observatorium side.
+    * We are missing other essential APIs for external user self-service e.g Tenancy, Retention Settings, Storage settings or Quotas
+  * Harder self-service of functionalities in case of missing feature (e.g lack of dashboarding or anything else that will come up)
+    * Mitigation: We can still do that by having a solid API exposed.
+  * Changing the functionalities for AppSRE, MTSRE of what they got used to: Lack of local alerting and local querying/dashboarding.
+
+### Shared Monitoring Stack
+
+In this alternative, we would deploy a single Prometheus instance for all managed tenants.
+
+![drawing](https://docs.google.com/drawings/d/e/2PACX-1vR95tWeFCXJDsVjORaJ1WlRurDEsdrHkoByEQreTxkly8KyWlD2bGbFI8WlJzIKm19j6g9zj7r8Lb8l/pub?w=1221&h=888)
+
+Pros:
+
+* We reduce the resource overhead for running Prometheus
+* Lower configuration and operational complexity.
+
+Cons:
+
+* Addons usage can impact each other (lack of QoS)
+* Tenancy isolation is easier with dedicated setups (e.g one isolated remote write)
+* We cannot set a different retention period (or any other global setting) per managed service. This would be useful if we decide to gradually shift more data to RHOBS.
+* The stack becomes more coupled and less composable due to its monolithic nature. It would essentially be a copy of User Workload Monitoring. If a third use-case comes up, we might need to build a new stack from scratch again.
+* With single (or HA) Prometheus, this stack will not scale. In order to scale we would need to deploy Observatorium which is [explained](#full-forwarding-mode-with-local-external-or-hosted-observatorium) in a previous alternative.
+* The same topology will be reproducible with the proposed alternative by specifying multiple namespaces in the `MonitoringStack` resource.
+
+### Shared Thanos Querier
+
+Pros:
+
+* Less resource overhead
+* Global view out of the box
+
+Cons:
+
+* This approach makes a `MonitoringStack` coupled to a centralized component. In order for individual teams to query their own metrics, they will need to query all prometheus instances.
+* Currently Thanos Query does not have good QoS mechanisms for independent query targets. Intensive queries from one team could therefore impact querying capabilities of other teams.
+* Since we want to build a community around the project, this approach makes the stack overall less composable and less reusable. It assumes that `MonitoringStack` owners are fine with exposing their metrics to each other. This might be the case for Managed Services, but it does not necessarily apply in the general sense.
+* In case a central view is needed, an additional Thanos Query instance can still be deployed independently.
```

Comment:
I think this will be a much better UX from the SRE perspective, although I understand the resource/RBAC constraints. Is this something we'd like to add to the long term considerations? 

I can imagine a good way to deploy such a querier would be through the operator, so that any new `MonitoringStack` Prometheis are added automatically as targets to the querier.

Reply:
Do we see a future where not all addons will be managed by the same SRE team? IOW, could we have MTSRE-1 responsible for addons 1, 2 and 3, and MTSRE-2 responsible for addons 4, 5, and 6?

Reply:
I think anyone can add Thanos Querier on top on their own (we can add that docs,) plus we talked about shared Grafana instances that will allow you to at least combine multiple sources under single dashboard, so it should be good experience (: